packages/x-date-pickers/src/EnhancedPickersDay/EnhancedPickersDay.tsx

```diff
@@ -0,0 +1,429 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { alpha, styled, useThemeProps, CSSInterpolation, Theme } from '@mui/material/styles';
+import ButtonBase from '@mui/material/ButtonBase';
+import {
+  unstable_useEnhancedEffect as useEnhancedEffect,
+  unstable_composeClasses as composeClasses,
+  unstable_useForkRef as useForkRef,
+} from '@mui/utils';
+import { DAY_MARGIN } from '../internals/constants/dimensions';
+import {
+  enhancedPickersDayClasses,
+  EnhancedPickersDayClassKey,
+  getEnhancedPickersDayUtilityClass,
+} from './enhancedPickersDayClasses';
+import { useUtils } from '../internals/hooks/useUtils';
+import { EnhancedPickersDayOwnerState, EnhancedPickersDayProps } from './EnhancedPickersDay.types';
+import { usePickerDayOwnerState } from '../PickersDay/usePickerDayOwnerState';
+import '@mui/system';
+
+const DAY_SIZE = 36;
+
+const useUtilityClasses = (ownerState: EnhancedPickersDayOwnerState) => {
+  const {
+    isDaySelected,
+    disableHighlightToday,
+    isDayCurrent,
+    isDayDisabled,
+    isDayOutsideMonth,
+    isDayFillerCell,
+  } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      isDaySelected && !isDayFillerCell && 'selected',
+      isDayDisabled && 'disabled',
+      !disableHighlightToday && isDayCurrent && !isDaySelected && !isDayFillerCell && 'today',
+      isDayOutsideMonth && 'dayOutsideMonth',
+      isDayFillerCell && 'fillerCell',
+    ],
+  };
+
+  return composeClasses(slots, getEnhancedPickersDayUtilityClass, {});
+};
+
+const overridesResolver = (
```

Comment:
`overridesResolver` and `styleArg` being only used in one place, I would just inline them.